### PR TITLE
add support for Rails prompt on v0.12+

### PR DESCRIPTION
**Pull request by r-obert**
I would have wished this could have been as simple as:
Pry::Prompt.add 'rails', 'description', PROMPTS

but because of how the public prompt API was implemented, it has
to be like this.

### CODE REVIEW

**Comment by r-obert** [comment]
@kyrylo @rf-

File: lib/pry-rails/prompt.rb [hunk 1]
```diff
@@ -22,20 +22,27 @@ def project_name
...
-  Pry::Prompt::MAP["rails"] = {
-    value: RAILS_PROMPT,
-    description: "Includes the current Rails environment and project folder name.\n" \
-                 "[1] [project_name][Rails.env] pry(main)>"
-  }
+  desc = "Includes the current Rails environment and project folder name.\n" \
+          "[1] [project_name][Rails.env] pry(main)>"
+  if Pry::Prompt.respond_to?(:add)
+    Pry::Prompt.add 'rails', desc, %w(> >>) do |target, nesting, _pry_, sep|
+      sep == '>' ? RAILS_PROMPT[0].call(target, nesting, _pry_) : RAILS_PROMPT[1].call(target, nesting, _pry_)
```
Comment (kyrylo) [inline lib/pry-rails/prompt.rb]: We can inline the proc and delete the `RAILS_PROMPT` constant:

```suggestion
"[#{pry.input_ring.size}] " \
  "[#{Prompt.project_name}][#{Prompt.formatted_env}] " \
  "#{pry.config.prompt_name}(#{Pry.view_clip(target_self)})" \
  "#{":#{nest_level}" unless nest_level.zero?}#{sep} "
```

Comment (r-obert) [reply lib/pry-rails/prompt.rb]: that's much better, thank you
rebased and pushed.

**Comment by rf-** [comment]
OK, I just released 0.3.7 with this PR plus some other small fixes.

**Comment by falak** [comment]
Just do
`bundle update pry-rails`
will remove the warning.

**Comment by pdobb** [comment]
The documentation seems to need an update related to this change. It says that, to get the rails prompt do:
```
if defined?(PryRails::RAILS_PROMPT)
  Pry.config.prompt = PryRails::RAILS_PROMPT
end
```

But this gem no longer defines `PryRails::RAILS_PROMPT`. I'm trying to sort out what is the right way to enable the prompt now. Will update if I figure it out.

**Pull request merged**